### DISCUSSION TITLE
Built-in worker loader should work with SharedWorker

### DISCUSSION THREAD

**Discussion by budleigh** [webpack/webpack#14986] [General]
## Feature request

**What is the expected behavior?**
The latest built-in loader for `Worker`s ([documented here](https://webpack.js.org/guides/web-workers/)) should also work when `Worker` is replaced with a `SharedWorker`. At the moment, a `SharedWorker` module isn't loaded at all.

**What is motivation or use case for adding/changing the behavior?**
Similar to the motivation for adding the built-in `Worker` loader in the latest version.

**How should this be implemented in your opinion?**
Likely in a similar fashion to the `Worker` loader, but I'm not super familiar with webpack internals.

**Are you willing to work on this yourself?**
yes

**Comment by alexander-akait** [comment]
It works https://github.com/webpack/webpack/blob/main/lib/dependencies/WorkerPlugin.js#L40, do not ignore how we can reproduce the problem if you need help
